internal/pkg/cli/env_delete_test.go

```diff
@@ -89,60 +173,51 @@ func TestDeleteEnvOpts_Ask(t *testing.T) {
 	}
 }
 
-func TestDeleteEnvOpts_Validate(t *testing.T) {
+func TestDeleteEnvOpts_Execute(t *testing.T) {
+	// Transform initDeleteEnvClientsFromProfile to a no-op while testing.
+	var oldInit = initDeleteEnvClientsFromProfile
+	initDeleteEnvClientsFromProfile = func(o *DeleteEnvOpts) error {
+		return nil
+	}
+	defer func() { initDeleteEnvClientsFromProfile = oldInit }()
+
 	const (
-		testProjName       = "phonetool"
-		testEnvName        = "test"
-		testRegion         = "us-west-2"
-		testManagerRoleARN = "arn:aws:iam::1111:role/phonetool-test-EnvManagerRole"
+		testProject = "phonetool"
+		testEnv     = "test"
 	)
-	var storeWithEnv = func(ctrl *gomock.Controller) *mocks.MockEnvironmentStore {
-		envStore := mocks.NewMockEnvironmentStore(ctrl)
-		envStore.EXPECT().GetEnvironment(testProjName, testEnvName).Return(&archer.Environment{
-			Project:        testProjName,
-			Name:           testEnvName,
-			Region:         testRegion,
-			ManagerRoleARN: testManagerRoleARN,
-		}, nil)
-		return envStore
-	}
+	testError := errors.New("some error")
 
 	testCases := map[string]struct {
-		inProjectName string
-		inEnv         string
-		mockStore     func(ctrl *gomock.Controller) *mocks.MockEnvironmentStore
-		mockRG        func(ctrl *gomock.Controller) *climocks.MockresourceGetter
+		inSkipPrompt bool
+		mockRG       func(ctrl *gomock.Controller) *climocks.MockresourceGetter
+		mockPrompt   func(ctrl *gomock.Controller) *climocks.Mockprompter
+		mockProg     func(ctrl *gomock.Controller) *climocks.Mockprogress
+		mockDeploy   func(ctrl *gomock.Controller) *climocks.MockenvironmentDeployer
+		mockStore    func(ctrl *gomock.Controller) *mocks.MockEnvironmentStore
```

Comment:
I prefer to have a single `setupMocks` func that initializes the mocks since it enables us to assert about ordering of calls if desired and doesn't require more mock functions as the dependency set grows.